### PR TITLE
Add shim layer for Cloudera CDS 3.3

### PR DESCRIPTION
**Pull request by sririshindra**
* Add shim layer for Cloudera CDS 3.3

### CODE REVIEW

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
build

File: jenkins/spark-premerge-build.sh [hunk 1]
```diff
@@ -57,6 +57,10 @@ mvn_verify() {
     $MVN_INSTALL_CMD -DskipTests -Dbuildver=322
     env -u SPARK_HOME $MVN_CMD -U -B $MVN_URM_MIRROR -Dbuildver=330 clean install $MVN_BUILD_ARGS \
       -Dpytest.TEST_TAGS='' -pl '!tools'
+
+    env -u SPARK_HOME $MVN_CMD -U -B $MVN_URM_MIRROR -Dbuildver=330cdh clean install $MVN_BUILD_ARGS \
```
Comment (tgravescs) [inline jenkins/spark-premerge-build.sh:61]: can you move this up to by the 321cdh one and just make it:

$MVN_INSTALL_CMD -DskipTests -Dbuildver=330cdh

File: sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcCastingShims.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+package com.nvidia.spark.rapids.shims
+
+import ai.rapids.cudf.{ColumnView, DType}
+import com.nvidia.spark.rapids.GpuOrcScan
+
+
+object OrcCastingShims {
+
+  def castIntegerToTimestamp(col: ColumnView, colType: DType): ColumnView = {
+    // For spark-321cdh, it consider the integers in `col` as milliseconds
```
Comment (tgravescs) [inline sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcCastingShims.scala:26]: update the comment to be 330cdh

File: sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcShims.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+// 330+ ORC shims
+object OrcShims {
+  // ORC Reader of the 321cdh Spark has no close method.
+  // The resource is closed internally.
+  def withReader[V](r: Reader)(block: Reader => V): V = {
+    block(r)
+  }
+
+  // ORC Reader of the 321cdh Spark has no close method.
```
Comment (tgravescs) [inline sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcShims.scala:38]: simiklar here, update 321cdh

**Review by tgravescs — COMMENTED** [review]
Thanks for the PR, mostly just a few nits on comments.  it looks like the new 330cdh profile is missing from a few of the pom files.. like the tests module and the udf_compiler module.  If they built maybe they just didn't pull any of the dependencies we had to change elsewhere.

File: sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/rapids/shims/spark330cdh/RapidsShuffleInternalManager.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * A shuffle manager optimized for the RAPIDS Plugin For Apache Spark.
+ * @note This is an internal class to obtain access to the private
+ *       `ShuffleManager` and `SortShuffleManager` classes.
+ */
+class RapidsShuffleInternalManager(conf: SparkConf, isDriver: Boolean)
+  extends RapidsShuffleInternalManagerBase(conf, isDriver) {
+
+  override def makeBypassMergeSortShuffleWriter[K, V](
```
Comment (abellina) [inline sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/rapids/shims/spark330cdh/RapidsShuffleInternalManager.scala:33]: @sririshindra, this got merged: https://github.com/NVIDIA/spark-rapids/pull/6511, and as a result you don't need this override as now it is just done inside`RapidsShuffleInternalManagerBase.scala`. So we'll need to remove it from this PR (along with the imports).

Sorry for the inconvenience

File: dist/pom.xml [hunk 5]
```diff
@@ -192,6 +194,75 @@
                 </pluginRepository>
              </pluginRepositories>
         </profile>
+        <profile>
+            <id>release330cdh</id>
+            <activation>
+                <property>
```
[the remark this answers is not in the document]
Comment (sririshindra) [reply dist/pom.xml:200]: I am not sure if it is off. Looks like they all have 4 spaces. maybe I am missing something. Could you point it out? I will fix it in the next commit.

File: jenkins/spark-premerge-build.sh (hunk 1 above)
Comment (sririshindra) [reply jenkins/spark-premerge-build.sh]: Fixed in the latest commit.

File: sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcCastingShims.scala (hunk 2 above)
Comment (sririshindra) [reply sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcCastingShims.scala]: Done

File: sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcShims.scala [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.orc.{CompressionCodec, CompressionKind, DataReader, OrcConf, OrcFile, OrcProto, PhysicalWriter, Reader, StripeInformation, TypeDescription}
+import org.apache.orc.impl.{DataReaderProperties, OutStream, SchemaEvolution}
+import org.apache.orc.impl.RecordReaderImpl.SargApplier
+
+import org.apache.spark.sql.execution.datasources.orc.OrcUtils
+import org.apache.spark.sql.types.DataType
+
+// 330+ ORC shims
+object OrcShims {
+  // ORC Reader of the 321cdh Spark has no close method.
```
[the remark this answers is not in the document]
Comment (sririshindra) [reply sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcShims.scala]: done

File: sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcShims.scala (hunk 3 above)
Comment (sririshindra) [reply sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/OrcShims.scala]: done

File: sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/spark330cdh/SparkShimServiceProvider.scala [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims.spark330cdh
+
+import com.nvidia.spark.rapids.{ClouderaShimVersion, ShimVersion}
+
+object SparkShimServiceProvider {
+  val VERSION = ClouderaShimVersion(3, 3, 0, "3.3.7180")
+  // cdh version can have numbers after after 7170
```
[the remark this answers is not in the document]
Comment (sririshindra) [reply sql-plugin/src/main/330cdh/scala/com/nvidia/spark/rapids/shims/spark330cdh/SparkShimServiceProvider.scala]: Done, fixed in the latest commit.

File: sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/rapids/shims/spark330cdh/RapidsShuffleInternalManager.scala (hunk 4 above)
Comment (sririshindra) [reply sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/rapids/shims/spark330cdh/RapidsShuffleInternalManager.scala]: Done, Fixed in the latest commit.

**Comment by tgravescs** [comment]
build

**Review by tgravescs — APPROVED** [review]
Looks good pending jenkins run

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
@sririshindra please don't force push in the future, it makes reviewing what changed harder.

**Review by tgravescs — APPROVED** [review]

**Pull request merged**